utils/validateLSP4Metadata.ts

```diff
@@ -45,14 +45,66 @@ export const validateLsp4MetaData = (
   }
 }
 
-const validateIcon = (icon: any) => {
-  return icon.width && icon.url && icon.hash && icon.hashFunction && icon.height
+/**
+ * Validates if the given image object follows proper structure
+ * It checks for old format using `hash` property and
+ * new format using `verification` property.
+ *
+ * @param image - image object to be validated
+ * @returns - true if validation passes, false otherwise
+ */
+const validateImage = (image: any) => {
+  return (
+    (image.width &&
+      image.height &&
+      image.url &&
+      image.hash &&
+      image.hashFunction) ||
+    (image.width &&
+      image.height &&
+      image.url &&
+      image.verification &&
+      image.verification.data &&
+      image.verification.method)
+  )
```

Comment:
nice - maybe it can go also in https://github.com/lukso-network/lsp-smart-contracts-utils
(for Daniel)

Reply:
Oh nice I wasn't aware we have this repo. We should definiately move this validation functions there as they are very useful. I also see some functions could be used in wallet ([getProfileMetadata](https://github.com/lukso-network/lsp-smart-contracts-utils/blob/develop/src/LSP3/getProfileMetadata/getProfileMetadata.ts), [getDigitalAssetCreators](https://github.com/lukso-network/lsp-smart-contracts-utils/tree/develop/src/LSP4/getDigitalAssetCreators)). On the other side it miss functions to get LSP7 and LSP8 metadata is it planned?

cc @b00ste

Reply:
Yeah, the plan is to get as many utility functions (LSP related) that are created across projects and move them to this library. Think long term this will be very helpful for everyone.
If you have some ideas, please share them with me either via discord or issues in the repo, and I will start looking over that asap.

Reply:
Let's discuss in discord maybe